src/product.h

```diff
@@ -77,6 +77,9 @@ class Product : public Resource {
   /// Changes the product's package id
   virtual void ChangePackageId(int new_package_id = Package::unpackaged_id());
 
+  // Packages the product
+  std::vector<Product::Ptr> Package(Package::Ptr pkg); 
```

Comment:
This should be virtual
```suggestion
  virtual std::vector<Product::Ptr> Package(Package::Ptr pkg); 
```

Reply:
I guess this raises the problem that packaging returns different types in `Material`/`Product` than it does for `Resource`